File: workflow/plugins/go/convert/convert_parse_json.go
// Package convert provides type conversion workflow plugins.
package convert

import (
	"encoding/json"

	plugin "metabuilder/workflow/plugins/go"
)

// ParseJSON parses a JSON string to object.
func ParseJSON(runtime *plugin.Runtime, inputs map[string]interface{}) (map[string]interface{}, error) {
	str, ok := inputs["string"].(string)
	if !ok {
		return map[string]interface{}{"result": nil, "error": "string is required"}, nil
	}

	var result interface{}
	if err := json.Unmarshal([]byte(str), &result); err != nil {
		return map[string]interface{}{"result": nil, "error": err.Error()}, nil
	}

	return map[string]interface{}{"result": result}, nil
}

File: workflow/plugins/python/convert/convert_parse_json.py
"""Workflow plugin: parse JSON string."""
import json


def run(_runtime, inputs):
    """Parse JSON string to object."""
    text = inputs.get("text", "")

    try:
        result = json.loads(text)
        return {"result": result}
    except json.JSONDecodeError as e:
        return {"result": None, "error": str(e)}
